README.md

```diff
@@ -1,2 +1,121 @@
-# raincatcher
-Feedhenry Raincatcher Core Repository
+# RainCatcher
+Feedhenry RainCatcher Core Repository
+
+## About RainCatcher
+
+## Developing
+
+RainCatcher supports and is tested on Node.JS@4 and should run on latest LTS.
+
+Check your Node.JS version running `node -v`
+
+### Setup
+
+```bash
+git clone https://github.com/feedhenry-raincatcher/raincatcher-core.git
+cd raincatcher
+npm install -g lerna
+npm install
+lerna bootstrap
+```
+
+## Repository folder structure
+
+This repository contains many subpackages managed through [Lerna](https://lernajs.io/), and they're
+contained in the following directories:
+
+<dl>
+  <dt>client/</dt>
+  <dd>Front-end packages, from API clients to user interface modules</dd>
+
+  <dt>cloud/</dt>
+  <dd>Cloud app packages, intended to be run on an Node.JS environment</dd>
+
+  <dt>demo/</dt>
+  <dd>Full-fledged Demo applications, showcasing the usage of multiple modules</dd>
+
+  <dt>examples/</dt>
+  <dd>Smaller examples and templates for other packages in the repository</dd>
+</dl>
+
+## Creating a new package
+
+In order to create a new package, we recommend duplicating
+[`examples/base`](./examples/base/README.md), which is a sample base that contains the skeleton
+expected of a new package. Refer to the linked README for more details.
+
+## Package structure
+
+### Directories
+
+<dl>
+  <dt>src/</dt>
+  <dd>Contains the TypeScript sources.</dd>
+
+  <dt>test/</dt>
+  <dd>The unit tests for the module. Should contain a `mocha.opts` file for configuring Mocha.</dd>
+
+  <dt>coverage/ and .nyc_output</dt>
+  <dd>Istanbul output, should be .gitignored.</dd>
+</dl>
+
+### Module structure
+
+RainCatcher modules are written in [TypeScript](http://typescriptlang.org).
+
+The main files are modules that by default export a class containing the main implementation for the module's intent,
+with a named export containing a public interface that should be depended upon, and reimplemented.
+
+```typescript
+import Implementation, { PublicInterface } from '@raincatcher/module';
+```
+
+See the [Reusing unit tests](#reusing-unit-tests) section in order to know how to leverage the existing test suites on
+custom implementations.
+
+#### Default exports
+
+For each module, the default export is the main implementation itself, as the most common scenario is it being used by
```

Comment:
As "in the". Best to avoid really long sentences. Keep it simple:
"Each module exports by default an implementation and set of interfaces by name."
Default implementation will usually implement most common use case for module.